File: src/main/java/me/cjcrafter/armormechanics/listeners/WeaponMechanicsDamageListener.java
package me.cjcrafter.armormechanics.listeners;

import me.cjcrafter.armormechanics.ArmorMechanicsAPI;
import me.deecaad.weaponmechanics.weapon.weaponevents.WeaponDamageEntityEvent;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;

public class WeaponMechanicsDamageListener implements Listener {

    @EventHandler (ignoreCancelled = true)
    public void onDamage(WeaponDamageEntityEvent event) {
        double damage = event.getFinalDamage();
        double rate = 1.0 - ArmorMechanicsAPI.getBulletResistance(event.getVictim().getEquipment(), event.getWeaponTitle());
        event.setFinalDamage(damage * rate);
    }
}

File: src/main/java/com/cjcrafter/armormechanics/listeners/WeaponMechanicsDamageListener.kt
package com.cjcrafter.armormechanics.listeners

import com.cjcrafter.armormechanics.ArmorMechanicsAPI
import com.cjcrafter.armormechanics.events.ResistBulletDamageEvent
import me.deecaad.weaponmechanics.weapon.weaponevents.WeaponDamageEntityEvent
import org.bukkit.Bukkit
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener

class WeaponMechanicsDamageListener : Listener {
    @EventHandler(ignoreCancelled = true, priority = EventPriority.HIGH)
    fun onDamage(event: WeaponDamageEntityEvent) {
        val damage = event.finalDamage

        val effects = ArmorMechanicsAPI.getBonusEffects(event.victim)
        var rate = 1.0
        for (effect in effects) {
            rate -= if (event.isExplosion)
                effect.getExplosionResistance(event.weaponTitle)
            else
                effect.getBulletResistance(event.weaponTitle)
        }

        val resistEvent = ResistBulletDamageEvent(event, rate)
        Bukkit.getPluginManager().callEvent(resistEvent)
        event.finalDamage = damage * resistEvent.rate
    }
}
